### PR TITLE
Add deprecation notes and warning codes

### PR DESCRIPTION
**Pull request by kuenishi**

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -8,9 +8,11 @@
 - Changed default value of `gc_max_workers` from 5 to 2 with its name
   changed to `gc.max_workers` with migration to config format change.
 
-## Notes on upgrading
+## Deprecation Notice
 
-- Changed webmachine's access log handler module name.
+- Replication on top of v2 replication has been deprecated.
```
Comment (shino) [inline RELEASE-NOTES.md]: `s/Replication/Multi-Datacenter Replication/`

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -8,9 +8,11 @@
...
+- Old list objects which required `fold_objects_for_list_keys` as
+  `false` *will be deprecated* at next version.
```
Comment (shino) [inline RELEASE-NOTES.md]: What does "deprecated" here mean?
`s/deprecated/removed/` ?

Comment (shino) [inline RELEASE-NOTES.md]: "next version" may be ambiguous, 2.0.1 or 2.1.0 or 3.0.0.

File: src/riak_cs_config.erl [hunk 3]
```diff
@@ -103,6 +104,24 @@
...
+-define(NOTICE(Bool, Msg),
+        case (Bool) of
+            true -> _ = lager:warning((Msg));
+            _ -> ok
+        end).
+
+-spec notices() -> ok.
+notices() ->
+    ?NOTICE(not riak_cs_list_objects_utils:fold_objects_for_list_keys(),
+            "`fold_objects_for_list_keys` will be deprecated"
```
Comment (shino) [inline src/riak_cs_config.erl]: removed?

File: src/riak_cs_config.erl [hunk 4]
```diff
@@ -103,6 +104,24 @@
...
+            " at next major version."),
+    ?NOTICE(anonymous_user_creation(),
+            "`anonymous_user_creation` is set as true. Set this as false"
+            " when this CS nodes is populated as public service."),
+    ?NOTICE(not gc_paginated_indexes(),
+            "`gc_paginated_indexes` is set as false"),
```
Comment (shino) [inline src/riak_cs_config.erl]: This config item is not in "Deprecation Notice" section of release note.

Comment (shino) [inline src/riak_cs_config.erl]: Ah, this is not deprecated, right?

Comment (kuenishi) [inline src/riak_cs_config.erl]: I changed my mind and this is also deprecated :)

**Comment by kuenishi** [comment]
Updated to comments.

File: RELEASE-NOTES.md [hunk 5]
```diff
@@ -8,9 +8,15 @@
...
 
-## Notes on upgrading
+## Deprecation Notice
 
-- Changed webmachine's access log handler module name.
+- Multi-Datacenter Replication on top of v2 replication support has
+  been deprecated.
+- Old list objects which required `fold_objects_for_list_keys` as
+  `false` is deprecated and *will be removed* at next major version.
+- Old GC pagination method used in case where `gc_paginated_indexes`
```
Comment (shino) [inline RELEASE-NOTES.md]: Old GC does not use pagination. For example, "Old non-paginated GC" seems nice.

File: src/riak_cs_config.erl [hunk 6]
```diff
@@ -103,6 +104,25 @@
 -include("s3_api.hrl").
 -include("list_objects.hrl").
 
+-define(NOTICE(Bool, Msg),
```
Comment (shino) [inline src/riak_cs_config.erl]: `notice` is another log severity level than `warning` in `lager` (even it is rarely used).
It seems a little strange for me that `NOTICE` macro uses `warning` severity.

**Comment by shino** [comment]
All changes looks nice :+1:

**Comment by kuenishi** [comment]
@borshop merge

**Pull request merged**